mllib/src/test/scala/org/apache/spark/ml/util/MLTest.scala

```diff
@@ -119,9 +119,15 @@ trait MLTest extends StreamTest with TempDirectory { self: Suite =>
     expectedMessagePart : String,
     firstResultCol: String) {
 
+    def hasExpectedMessageDirectly(exception: Throwable): Boolean =
+      exception.getMessage.contains(expectedMessagePart)
+
     def hasExpectedMessage(exception: Throwable): Boolean =
-      exception.getMessage.contains(expectedMessagePart) ||
-        (exception.getCause != null && exception.getCause.getMessage.contains(expectedMessagePart))
+      hasExpectedMessageDirectly(exception) || (
+        exception.getCause != null && (
+          hasExpectedMessageDirectly(exception.getCause) || (
+            exception.getCause.getCause != null &&
+            hasExpectedMessageDirectly(exception.getCause.getCause))))
```

Comment:
Why not use a loop to check inner exception ? So it can check any deep level exception msg.
Maybe we cannot make sure the possible root exception max level (even in master version)
@jkbradley What do you think ?

Reply:
I don't think we need to loop further.  If the real message is buried this deep, that could be considered a problem in and of itself.